packages/components/src/validated-form-controls/control-with-error.tsx

```diff
@@ -111,13 +116,58 @@ function UnforwardedControlWithError< C extends React.ReactElement >(
 		};
 	} );
 
-	const validate = () => {
-		const message = customValidator?.();
+	useEffect( () => {
+		if ( ! isTouched ) {
+			return;
+		}
+
 		const validityTarget = getValidityTarget();
 
-		validityTarget?.setCustomValidity( message ?? '' );
-		setErrorMessage( validityTarget?.validationMessage );
-	};
+		if ( ! customValidityMessage?.type ) {
+			validityTarget?.setCustomValidity( '' );
+			setErrorMessage( validityTarget?.validationMessage );
+			setStatusMessage( undefined );
+			return;
+		}
+
+		switch ( customValidityMessage?.type ) {
+			case 'validating': {
+				// Wait before showing a validating state.
+				const timer = setTimeout( () => {
```

Comment:
I wonder if this should be using some sort of shared ref for the timer and assigning as `timerRef.current ??= ...` ? If somehow this `useEffect` were triggered without any meaningful changes (e.g. new `getValidityTarget` reference because it's defined as an inline function) then the timer could reset unnecessarily.